crates/turborepo-lib/src/config/override_env.rs

```diff
@@ -0,0 +1,493 @@
+#[allow(non_snake_case)]
+struct Input {
+    TURBO_TEAM: Option<String>,
+    TURBO_TEAMID: Option<String>,
+    TURBO_TOKEN: Option<String>,
+    VERCEL_ARTIFACTS_OWNER: Option<String>,
+    VERCEL_ARTIFACTS_TOKEN: Option<String>,
+}
+
+impl Input {
+    fn new() -> Self {
+        Self {
+            TURBO_TEAM: None,
+            TURBO_TEAMID: None,
+            TURBO_TOKEN: None,
+            VERCEL_ARTIFACTS_OWNER: None,
+            VERCEL_ARTIFACTS_TOKEN: None,
+        }
+    }
+}
+
+impl<'a> From<&'a HashMap<OsString, OsString>> for Input {
+    fn from(environment: &'a HashMap<OsString, OsString>) -> Self {
+        Self {
+            TURBO_TEAM: environment
+                .get(OsStr::new("turbo_team"))
+                .map(|s| s.to_str().unwrap().to_string()),
+            TURBO_TEAMID: environment
+                .get(OsStr::new("turbo_teamid"))
+                .map(|s| s.to_str().unwrap().to_string()),
+            TURBO_TOKEN: environment
+                .get(OsStr::new("turbo_token"))
+                .map(|s| s.to_str().unwrap().to_string()),
+            VERCEL_ARTIFACTS_OWNER: environment
+                .get(OsStr::new("vercel_artifacts_owner"))
+                .map(|s| s.to_str().unwrap().to_string()),
+            VERCEL_ARTIFACTS_TOKEN: environment
+                .get(OsStr::new("vercel_artifacts_token"))
+                .map(|s| s.to_str().unwrap().to_string()),
+        }
+    }
+}
+
+// this is an internal structure (that's a partial of ConfigurationOptions) that
+// we use to store
+pub struct Output {
+    /// maps to ConfigurationOptions.team_id
+    pub team_id: Option<String>,
+    // maps to ConfigurationOptions.team_slug
+    pub team_slug: Option<String>,
+    // maps to ConfigurationOptions.token
+    pub token: Option<String>,
+}
+
+impl Output {
+    fn new() -> Self {
+        Self {
+            team_id: None,
+            team_slug: None,
+            token: None,
+        }
+    }
+}
+
+// get Output from Input
+impl From<Input> for Output {
+    fn from(input: Input) -> Self {
+        let mut output = Output::new();
+
+        // TURBO_TEAMID+TURBO_TOKEN
+        if input.TURBO_TEAMID.is_some() && input.TURBO_TOKEN.is_some() {
+            output.team_id = input.TURBO_TEAMID;
+            output.token = input.TURBO_TOKEN;
+
+            if input.TURBO_TEAM.is_some() {
+                // there can also be a TURBO_TEAM, so we'll use that as well
+                output.team_slug = input.TURBO_TEAM;
+            }
+
+            return output;
+        }
+
+        // TURBO_TEAM+TURBO_TOKEN
+        if input.TURBO_TEAM.is_some() && input.TURBO_TOKEN.is_some() {
+            output.team_slug = input.TURBO_TEAM;
+            output.token = input.TURBO_TOKEN;
+
+            if input.TURBO_TEAMID.is_some() {
+                // there can also be a TURBO_TEAMID, so we'll use that as well
+                output.team_id = input.TURBO_TEAMID;
+            }
+
+            return output;
+        }
```

Comment:
![Screenshot_20241016_210935](https://github.com/user-attachments/assets/64bd9cd0-22ad-4e24-a012-13f1911a2a39)

They are in fact subtly different.  I personally have to read at like 0.0001 words per minute to catch it.  Open to ideas if you think there's a better way to restructure these sections.  I sorta intentionally did them in a really "stupid/simple" way because there's just too much tricky stuff here with the business logic in the first place

Reply:
Ah, you're right!

You could use a match statement on a tuple of all the input values, but that has the downside of losing the field names. Up to you if `(None, Some(_TURBO_TEAM_ID), None, Some(_TURBO_TOKEN)) => ` would be easier to understand compared to the if's